src/entity/posts/Post.ts

```diff
@@ -136,4 +144,25 @@ export class Post {
 
   @Column({ default: false })
   private: boolean;
+
+  @Column({ default: true })
+  visible: boolean;
+
+  @Column({ default: null })
+  visibleAt: Date;
+
+  @Column({ default: null, type: 'text' })
+  private origin: PostOrigin;
+
+  @BeforeInsert()
+  @BeforeUpdate()
+  private setOrigin(): void {
+    if (this.sourceId === COMMUNITY_PICKS_SOURCE) {
+      this.origin = PostOrigin.CommunityPicks;
+    } else if (this.type === PostType.Share) {
+      this.origin = PostOrigin.Squads;
+    } else {
+      this.origin = PostOrigin.Crawler;
+    }
+  }
```

Comment:
I don't think we should rely on this.
Maybe it's best to have it being either set on creation or updated through the update hook (still coming)

Maybe we can say `crawler` is the default though?
@idoshamun wdyt?

Reply:
Setting it as default is a good point since we have it on `else` already.

Reply:
When thinking about it, I believe `origin` should be optional. In the case of share post, there's no origin. Or if we want to set it to something we need to call it `ugc` or something like that. We don't want to trigger Yggdrasil for share posts.
sidenote: We've had a lot of issues with `@BeforeInsert` and `@BeforeUpdate` in the past. I suggest setting the field directly in the right place instead of relying on the source

Reply:
@rebelchris do you have any recommendations on what we should use for the `Share` type?

Reply:
What does UGC stand for in this context?

Reply:
I think it is "User-generated content".

Reply:
Hmm, but private links will also be UGC then.

Since we are the ones publishing the topic to yggdrasil, maybe we should be more explicit.

- PrivateLink (for new private links content) (this needs to be send)
- SharedPost (for the actual shares, these will be excluded from sending, even if they are not visible yet.

@idoshamun What do you think?

Reply:
Private links shouldn't not be `ugc`, they should be `squads`.
Share post will get `ugc` or even null if we want. I think it might be better.

And again I'm not a fan of @BeforeInsert() and @BeforeUpdate(). I prefer to set it explicitly when needed

Reply:
Should we not care if it's simply a sharedPost or new original private link in a squad?
We need to know the difference because for private links we will notify users where only the origin will be able to tell the difference.

Reply:
As we agreed, upon the submission of a private link two posts will be created:
article post with the `squads` origin that will trigger yggdrasil
shared post that refers to the above with `ugc` or null (I'm fine with both) origin that shouldn't trigger yggdrasil.
Both will be invisible until yggdrasil finishes

Reply:
Yes we have the same idea, I'm not not a fan of using UGC as technically both are user generated.
Why not be more explicit about which part is which?

My only concern is naming.

Reply:
This is an outdated change, the latest commit is not using the `Before` decorators anymore. Either label works for me. Let me know what you think suits best.